Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -139,10 +139,22 @@ void HotkeyScheduler::Run()
 
   while (!m_stop_requested.IsSet())
   {
-    Common::SleepCurrentThread(1000 / 60);
+    const u64 freelook_update_freq = 200;
```

Comment:
I'd move this to a constexpr above, together with the hotkeys 60Hz update rate (I've also done the same change, it's nicer and highlights that there shouldn't be any negative consequence in changing them).

Reply:
Why is the refresh rate 200Hz? Maybe it should be hinted. Is it because you want the input polling to be as up to date as possible when a new render starts? Or can freelook actually output more frames that the game naturally would (I don't know how it works)?

Reply:
I spent a very long time getting it to match master (you can see all the comments about it not feeling right by users).  Therefore I'm hesitant to change much related to the timing.  The previous timings were from Qt events.  I can't recall where I found the value.  But your other point makes me think it's not really working as I intended.  I will investigate.

Reply:
The only advantage in updating at a higher rate than the game would be to have more up to date values, which is really important if you are wearing a headset (if that is possible), otherwise this is likely useless and was only done to increase the accuracy of fake relative inputs.

Reply:
I still need to test but I believe I picked a value of 200 likely to match Wiimote which I believe is [also updated at 200hz](https://github.com/dolphin-emu/dolphin/blob/0029ca84b07f44e3ce8c80a44fa0285a966b34d3/Source/Core/Core/HW/Wiimote.h#L78) and also because less frequent  updates yielded less accurate motion results.

Reply:
Wiimote BT inputs are at 200 (I guess fixed, independently of the emulation speed).
Why would less frequent updates have less accurate motion results if you are just reading the current pitch/yaw/roll value?

Reply:
DSU is angular velocity.  The values passed are relative motion, not absolute, so I believe the input rate is important to get that smooth feel.

Reply:
Sure, if you are accumulating these values to move the camera within that update in the hotkey thread, then it makes sense.
Just add a comment explaining that's why it's 200 then.

Reply:
Yeah. Ideally angular velocity values would be processed as their source device produces them. Googling implies DS4 controllers support 1000Hz over Bluetooth. It's a shame to only process gyro data at 200hz and not at all in sync with device updates.

The least intrusive solution I can think of would be to add gyro/accel processing at the input backend level. Absolute rotation angle inputs can then be provided.

Reply:
Exactly. Little code involved, cleaner implementation, more accurate.

Reply:
I'm fine with something like that but that is a problem beyond this PR imo.

Also, I worry a bit about your assumptions.  The DSU is provider specific.  For instance, my OpenTrack DSU implementation does not happen at the rate of the DS4.  In particular, it listens over UDP to OpenTrack, so it's whatever rate that program provides.  But that is just one provider and anyone is free to create their own.  I suppose we could make it configurable but eh.

Reply:
Just use the delta time between packets. Or try to deduce the frequency at which they are sent and then use that number. It won't be any less accurate than it is now.

Reply:
If there's work to be improved on the input side, that's fine but that's not really the kind of thing I'm interested in working on personally.  And if I'm being honest, I wouldn't have any idea where to begin.  Multiple people, with different devices, have stated that the current implementation works fine.  This may be sub-optimal but I don't feel like such a solution needs to hold up _this_ review.

Reply:
(I'll just add, the input object is used for multiple devices: keyboard/mouse, gamepad, dsu, etc..so one frequency is not going to cut it.  It'd have to be the highest frequency from all the inputs because each input could hypothetically have its own frequency rate)